### PR TITLE
FEATURE: Parse images in email signatures

### PR DESCRIPTION
**Pull request by nbianca**

### CODE REVIEW

File: app/services/inline_uploads.rb [hunk 1]
```diff
@@ -197,19 +197,25 @@ def self.match_anchor(markdown, external_href: false)
     end
   end
 
-  def self.match_img(markdown, external_src: false)
+  def self.match_img(markdown, external_src: false, uploads: nil)
     markdown.scan(/(<(?!img)[^<>]+\/?>)?(\s*)(<img [^>\n]+>)/i) do |match|
-      node = Nokogiri::HTML5::fragment(match[2].strip).children[0]
+      # rubocop:disable Discourse/NoNokogiriHtmlFragment
+      node = Nokogiri::HTML::fragment(match[2].strip).children[0]
```
Comment (ZogStriP) [inline app/services/inline_uploads.rb:203]: Why was this change required?

File: lib/email/receiver.rb [hunk 2]
```diff
@@ -1100,21 +1102,28 @@ def notify_about_rejected_attachment(attachments)
...
     end
 
-    def create_post(options = {})
-      options[:via_email] = true
-      options[:raw_email] = @raw_email
-
-      options[:created_at] ||= @mail.date
-      options[:created_at] = DateTime.now if options[:created_at] > DateTime.now
-
+    def add_elided_to_raw!(options)
```
Comment (ZogStriP) [inline lib/email/receiver.rb:1105]: This method is weird. It's returning `options` which looks like is never used and also updates "in-place" the `options` argument?

File: app/services/inline_uploads.rb (hunk 1 above)
Comment (nbianca) [reply app/services/inline_uploads.rb]: I just tried this one again and it looks like it was a problem with the example I used for testing. In my development environment, that email was parsed differently by `Nokogiri::HTML` and `Nokogiri::HTML5`. The change is no longer needed. :blush:

File: lib/email/receiver.rb (hunk 2 above)
Comment (nbianca) [reply lib/email/receiver.rb:1105]: You are right, I was torn between the two and finally went with the in-place changes and left the return code too.

**Comment by LinqLover** [comment]
Just a quick note from a stranger who just discovered this patch by accident: Your fixture <https://raw.githubusercontent.com/discourse/discourse/main/spec/fixtures/emails/body_with_image.eml> is not compliant to the specs. If you want to embed inline resources from a MIME document via multipart, you should use `multipart/related` instead of `multipart/mixed`. The semantics of `multipart/mixed` is to combine all resources (which could be a simple concatenation of texts and attachments), whereas `multipart/related` makes clear that the resources depend on each other. While the email clients you may have tested this with might display the message correctly, the fixture does not adhere to the standards. Just a note from a stranger person, maybe it helps you anyway. :-)
